From: Ritesh Raj Sarraf <rrs@debian.org>
Date: Tue, 30 Oct 2012 11:27:31 +0100
Subject: Fix build failure for header.py and typelib.py

Earlier builds were passing the --cachedir option
Sometime during TB15, Mozilla changed that to variables.
This change was not passing the --cachedir option, hence the build failure.
This patch just hacks the build by passing the cachedir option manually

---
 config/rules.mk |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config/rules.mk b/config/rules.mk
index cd988df..05100d4 100644
--- a/config/rules.mk
+++ b/config/rules.mk
@@ -1454,7 +1454,7 @@ $(XPIDL_GEN_DIR)/%.h: %.idl $(XPIDL_DEPS) $(XPIDL_GEN_DIR)/.done
 	$(REPORT_BUILD)
 	$(PYTHON) -u $(MOZILLA_DIR)/config/pythonpath.py \
 	  $(PLY_INCLUDE) \
-	  $(LIBXUL_DIST)/sdk/bin/header.py $(XPIDL_FLAGS) $(_VPATH_SRCS) -d $(MDDEPDIR)/$(@F).pp -o $@
+	  $(LIBXUL_DIST)/sdk/bin/header.py $(XPIDL_FLAGS) $(_VPATH_SRCS) -d $(MDDEPDIR)/$(@F).pp --cachedir=. -o $@
 	@if test -n "$(findstring $*.h, $(EXPORTS))"; \
 	  then echo "*** WARNING: file $*.h generated from $*.idl overrides $(srcdir)/$*.h"; else true; fi
 
@@ -1466,7 +1466,7 @@ $(XPIDL_GEN_DIR)/%.xpt: %.idl $(XPIDL_DEPS) $(XPIDL_GEN_DIR)/.done
 	$(PYTHON) -u $(MOZILLA_DIR)/config/pythonpath.py \
 	  $(PLY_INCLUDE) \
 	  -I$(MOZILLA_DIR)/xpcom/typelib/xpt/tools \
-	  $(LIBXUL_DIST)/sdk/bin/typelib.py $(XPIDL_FLAGS) $(_VPATH_SRCS) -d $(MDDEPDIR)/$(@F).pp -o $@
+	  $(LIBXUL_DIST)/sdk/bin/typelib.py $(XPIDL_FLAGS) $(_VPATH_SRCS) -d $(MDDEPDIR)/$(@F).pp --cachedir=. -o $@
 
 # no need to link together if XPIDLSRCS contains only XPIDL_MODULE
 ifneq ($(XPIDL_MODULE).idl,$(strip $(XPIDLSRCS)))
